addOns/automation/src/main/java/org/zaproxy/addon/automation/ExtensionAutomation.java

```diff
@@ -206,22 +210,30 @@ private AutomationProgress runPlan(LinkedHashMap<?, ?> envData, ArrayList<?> job
                             Constant.messages.getString("automation.error.job.data", paramsObj));
                     continue;
                 }
+                job.setVerify(true);
                 job.applyParameters((LinkedHashMap<?, ?>) paramsObj, progress);
-
-                // In case errors or warnings are encountered in the applyParameters() call above
-                if (env.isTimeToQuit()) {
-                    break;
-                }
-
-                progress.info(
-                        Constant.messages.getString("automation.info.jobstart", job.getType()));
-                job.runJob(env, jobData, progress);
-                progress.info(Constant.messages.getString("automation.info.jobend", job.getType()));
+                jobsToRun.put(job, jobData);
             } else {
                 progress.error(
                         Constant.messages.getString("automation.error.job.unknown", jobType));
             }
         }
+
+        if (env.isTimeToQuit()) {
+            return progress;
+        }
+
+        for (Entry<AutomationJob, LinkedHashMap<?, ?>> jobInfo : jobsToRun.entrySet()) {
+            AutomationJob job = jobInfo.getKey();
+            LinkedHashMap<?, ?> jobData = jobInfo.getValue();
+            Object paramsObj = jobData.get("parameters");
+            job.setVerify(false);
+            job.applyParameters((LinkedHashMap<?, ?>) paramsObj, progress);
```

Comment:
Why apply again?

Reply:
The initial call to applyParameters doesnt actually "apply" anything. With `verify` set to `true` we just check for all errors, and then we set it to false to actually apply the changes. This is done as someone might make an assumption on the initial state, which would be wrong if we apply all of the params first.

Reply:
Clarified from a talk in IRC.

Reply:
Missed your comment here.

Reply:
Unresolved this still needs to be discussed.

Reply:
From a quick chat in IRC, the issues that arise from parsing the values (e.g. `PassiveScanWaitJob` when using `maxDuration`) should also be detected when validating.

Reply:
Must admit I'm not so keen on the `verify` field. That could cause problems in the future if we ever decided to run jobs in parallel.
I realise that it was introduced in order to maintain backwards compatiblity but I'd rather have a `verify` parameter on a new `applyParameters` method. This could have a default implementation which does nothing if `verify` is true and calls the old method if its false. Or something along those lines anyway ;)

Reply:
Tweaked as required